File: LeetCode/Medium/150. Evaluate Reverse Polish Notation.py
class Solution:
    
    def evalRPN( self, tokens: List[str] ) -> int:
        
        nums = []
        
        for t in tokens:
            if t[-1].isdecimal():
                nums.append( int( t ) )
            else:
                o = nums.pop()
                
                if t == "*":
                    nums[-1] *= o
                elif t == "/":
                    nums[-1] = int( nums[-1] / o )
                elif t == "+":
                    nums[-1] += o
                else:
                    nums[-1] -= o
            
        return nums[0]

File: LeetCode/Medium/150. Evaluate Reverse Polish Notation.java
class Solution
{

    public int evalRPN
    ( String[] tokens )
    {
        Stack<Integer> stack =
            new Stack<>();
        Integer temp;

        for ( String token : tokens )
        {
            switch ( token )
            {
                case "+":
                    stack.push( stack.pop() + stack.pop() );
                    break;
                case "-":
                    temp = stack.pop();
                    stack.push( stack.pop() - temp );
                    break;
                case "*":
                    stack.push( stack.pop() * stack.pop() );
                    break;
                case "/":
                    temp = stack.pop();
                    stack.push( stack.pop() / temp );
                    break;
                default:
                    stack.push( Integer.parseInt( token ) );
            }
        }

        return stack.pop();
    }

}
